pkg/template_generator.go

```diff
@@ -130,6 +130,19 @@ func generateTemplateForPodSpec(podSpec kapi.PodSpec, key string, value map[stri
 		podSpec.RestartPolicy = kapi.RestartPolicy(fmt.Sprintf("{{.Values.%s.%s}}", key, RestartPolicy))
 	}
 
+	if len(podSpec.ImagePullSecrets) != 0 {
+		imagePullSecretsObj := []kapi.LocalObjectReference{}
+
+		for _, imagePullSecrets := range podSpec.ImagePullSecrets {
+			imagePullSecrets.Name = generateSafeKey(imagePullSecrets.Name)
+			value["imagePullSecrets"] = imagePullSecrets.Name
```

Comment:
Yes, secret yaml cannot be always part of chart. So, in that case i ll need to keep the name same everywhere, so no issue here.

For the scenario where i want to include secret yml in charts, the secret name need to be in sync in podSpec. In that case i think, the secret name in such scenario where there is internal dependency should not be touched. I mean generateSafeKey() should not be run on secret metadata name. This way it will ensure the name will be sync as required from original imput and not needed to be configurable. Will avoid putting in any complex logic for arrays and all.
WDYT?